app/Policies/PostPolicy.php

```diff
@@ -27,7 +27,8 @@ public function __construct()
      */
     public function viewAll($user, Post $post)
     {
-        if ($user === null) {
+        // If there is no user and it is not a client with admin credentials, return false.
+        if ($user === null && is_staff_user() === false) {
```

Comment:
Do you still need the `$user === null` part?

Reply:
@katiecrane great q! I think we want that in case it is an anonymous user.

Reply:
@chloealee wouldn't an anonymous user also not be a staff user?

Reply:
oh that is a good point, @katiecrane ! I wonder though if `$user->northstar_id === $post->northstar_id;` will break if there is no `$user` to try to get the `northstar_id` from

Reply:
Yeah, it might be good to separate these into two different cases for readability.

Reply:
Oh, although that doesn't work since then we'd prevent users from seeing their own info. Bleh.

Reply:
ah this is a lot more readable, thanks @DFurnes ! I just updated this :)

Just want to double check - is it safe to assume that all machines should have the `admin` scope, right? I see that they all have it now but just want to be sure this is safe to assume in the future too.

Reply:
I'm thinking it makes sense to remove that scope in the future, since it's confusing and possible to misuse. For Rogue, I think allowing any machine client to access full profiles is a-okay.